analyzer/src/main/kotlin/managers/CocoaPods.kt

```diff
@@ -46,7 +86,206 @@ class CocoaPods(
+            packages = packages,
+            project = Project(
+                id = Identifier(
+                    type = managerName,
+                    namespace = "",
+                    name = definitionFile.relativeTo(analysisRoot).invariantSeparatorsPath,
+                    version = ""
+                ),
+                definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                authors = sortedSetOf(),
+                declaredLicenses = sortedSetOf(),
+                vcs = VcsInfo.EMPTY,
+                vcsProcessed = processProjectVcs(workingDir),
+                scopeDependencies = scopes,
+                homepageUrl = ""
+            ),
+            issues = issues
+        )
+
+        return listOf(projectAnalyzerResult)
+    }
+
+    private fun getPackage(id: Identifier, workingDir: File): Package {
+        val podspec = getPodspec(id, workingDir)
+
+        val vcs = podspec.source["git"]?.let { url ->
+            VcsInfo(
+                type = VcsType.GIT,
+                url = url,
+                revision = podspec.source["tag"].orEmpty()
+            )
+        } ?: VcsInfo.EMPTY
+
+        return Package(
+            id = id,
+            authors = sortedSetOf(),
+            declaredLicenses = listOf(podspec.license).toSortedSet(),
+            description = podspec.summary,
+            homepageUrl = podspec.homepage,
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = podspec.source["http"]?.let { RemoteArtifact(it, Hash.NONE) } ?: RemoteArtifact.EMPTY,
+            vcs = vcs,
+            vcsProcessed = processPackageVcs(vcs, podspec.homepage)
+        )
+    }
+
+    private fun getPodspec(id: Identifier, workingDir: File): Podspec {
+        podspecCache[id.name]?.let { return it }
+
+        val podspecName = id.name.substringBefore("/")
+
+        val podspecFile = run("spec", "which", podspecName, "--version=${id.version}", "--allow-root",
+            "--regex", workingDir = workingDir
+        ).requireSuccess().stdout.trim().let { File(it) }
+
+        podspecFile.readValue<Podspec>().withSubspecs().associateByTo(podspecCache, { it.name })
+
+        return podspecCache.getValue(id.name)
+    }
+}
+
+private const val LOCKFILE_FILENAME = "Podfile.lock"
+
+private const val SCOPE_NAME = "dependencies"
+
+private val NAME_AND_VERSION_REGEX = "([\\S]+)\\s+(.*)".toRegex()
+
+private fun getPackageReferences(podfileLock: File): SortedSet<PackageReference> {
+    val versionForName = mutableMapOf<String, String>()
+    val dependenciesForName = mutableMapOf<String, MutableSet<String>>()
+    val root = yamlMapper.readTree(podfileLock)
+
+    root.get("PODS").asIterable().forEach { node ->
+        val entry = when (node) {
+            is ObjectNode -> node.fieldNames().asSequence().first()
+            else -> node.textValue()
+        }
+
+        val (name, version) = NAME_AND_VERSION_REGEX.find(entry)!!.groups.let {
```

Comment:
How can we be sure the `!!` in this line never causes an NPE?

Reply:
To my understanding the file format of the lockfile implies that `!!` does not cause an NPE.

Reply:
So you're saying we *should* throw an exception on malformed files / not be lenient by skipping invalid entries or so. Ok.

Reply:
I'd prefer adding that fault tolerance only in case need arises.